app/jobs/create_institution_records_job.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'net/http'
+
+class CreateInstitutionRecordsJob < ApplicationJob
```

Comment:
Jobs aren't perfect for storing business logic - they're rather hard to test, their logic is hard to reuse, and they are hard to refactor. It's rather customary for jobs to have a very minimal `perform` method and store the logic somewhere else - here, it'd probably be a service object since it's suitable for encapsulating side effects (example: https://github.com/Selleo/pattern/#service).
```ruby
def perform(institution_type_id)
  CreateInstitutionRecords.call(institution_type_id)
end
```
That way, it will be way easier for you to test `CreateInstitutionRecords` in isolation and reuse it in other places. You'll also be able to use instance variables and so do a proper object-oriented modeling

Reply:
I will design the data ingestion job from gdynia as a service object but lets leave these two as they are for now